src/resmom/linux/alps.c

```diff
@@ -3727,21 +3727,22 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 						goto bad_vnl;
 					if (accel->data.gpu->memory) {
 						attr = "resources_available.accelerator_memory";
-						if (totseg == 0) {
+						if (totseg == 0 || vnode_per_numa_node) {
 							snprintf(utilBuffer,
 								sizeof(utilBuffer),
 								"%umb",
 								accel->data.gpu->memory);
-						} else if (vnode_per_numa_node) {
-							snprintf(utilBuffer,
-								sizeof(utilBuffer),
-								"@%s_%ld_0",
-								mpphost, node->node_id);
+							if (vnode_per_numa_node && !totseg) {
+								snprintf(utilBuffer,
+									 sizeof(utilBuffer),
+									 "@%s_%ld_0",
+									 mpphost, node->node_id);
+							}
+							if (vn_addvnr(nv, vname, attr,
+								      utilBuffer, 0,
+								      0, NULL) == -1)
+								goto bad_vnl;
```

Comment:
Hi Lisa,
The old code was in a different function due to which that condition worked well. But for basil 1.3 we added a new function inventory_loop_on_segments to get resource value and the old part of the code was move to this new function which caused the issue, since now it was getting called multiple times for multiple segments.

Regards
Dilip